src/runtime/composables/useSession.ts

```diff
@@ -87,7 +87,13 @@ const signIn = async (
     return navigateToWithNuxt(errorUrl)
   }
 
-  // 2. Redirect to the general sign-in page with all providers in case either no provider or no valid provider was selected
+  // 2. Check if a custom provider was given and use its defaultProvider
+  const runtimeConfig = publicRuntimeConfig()
+  if (runtimeConfig?.auth?.defaultProvider !== undefined) {
+    provider = runtimeConfig.auth.defaultProvider
```

Comment:
we should never override the `provider` that was passed to us, we should only override it if `prover = undefined`  was passed